lucene/core/src/java/org/apache/lucene/util/automaton/Operations.java

```diff
@@ -1091,25 +1091,33 @@ public static String getCommonPrefix(Automaton a) {
    * @return common prefix, which can be an empty (length 0) BytesRef (never null)
    */
   public static BytesRef getCommonPrefixBytesRef(Automaton a) {
```

Comment:
Some optimizations in this method (always called to build the commonSuffix for CompiledAutomaton for infinite language, e.g. PrefixQuery and WildcardQuery). Lazy create structures (WildcardQuery may not have a first single-path state), and only one visited.add() call. This speeds up mainly WildcardQuery (a couple % in my luceneutil measures).